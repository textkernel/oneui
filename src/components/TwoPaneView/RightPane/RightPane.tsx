import * as React from 'react';
import { bem } from '../../../utils';
import styles from './RightPane.scss';

export interface Props extends React.HTMLAttributes<HTMLDivElement> {
    /** Node(s) to be rendered in the right pane */
    children?: React.ReactNode;
}

const { block } = bem('RightPane', styles);

export const RightPane: React.FC<Props> = (props) => {
    const { children, ...rest } = props;

    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

RightPane.displayName = 'RightPane';
