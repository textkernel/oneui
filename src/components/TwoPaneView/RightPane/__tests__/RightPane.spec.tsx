import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { RightPane } from '../RightPane';

describe('<RightPane> that renders a right pane of a two pane view', () => {
    it('should render correctly', () => {
        const view = render(<RightPane>Some content</RightPane>);

        expect(view.container).toMatchSnapshot();
        expect(view.container).toHaveTextContent('Some content');
    });
});
