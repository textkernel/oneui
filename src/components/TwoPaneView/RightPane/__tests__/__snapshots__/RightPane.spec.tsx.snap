// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightPane> that renders a right pane of a two pane view should render correctly 1`] = `
<div>
  <div
    class="RightPane"
  >
    Some content
  </div>
</div>
`;
